internal/handler/site/middleware/compression.go

```diff
@@ -0,0 +1,28 @@
+package middleware
+
+import (
+	"net/http"
+	"io"
+
+	"github.com/andybalholm/brotli"
+	"github.com/oursky/pageship/internal/site"
+)
+
+type CompressedResponseWriter struct {
+	http.ResponseWriter
+	compressor io.WriteCloser
+}
+
+func (crw CompressedResponseWriter) Write(b []byte) (int, error) {
+	defer crw.compressor.Close()
```

Comment:
`ResponseWriter.Write` could be called multiple times to form a single compressed data stream, so we should not call `Close` here. Instead, call `Close` after all `Write` is done, i.e. after `next.ServeHTTP`.